### DISCUSSION TITLE
Use different version of nixpkgs for nixvim

### DISCUSSION THREAD

**Discussion by gbrls** [nix-community/nixvim#1520] [Q&A]
Hey folks, is there a way to use nixvim unstable in a nixos using nixpkgs for 23.11?

I don't want to turn my whole system to use the unstable branch, but I want to have the latest version of nixvim since a few features that I'm interested are only available for unstable: (obsidian plugin, gleam support).

Thanks in advance :)

**Comment by MattSturgeon** [comment]
IMO the best way to do this is using nixvim's "standalone" mode.

In this mode, you can use a nixvim conifg to build a custom (wrapped) neovim package using whatever nixpkgs channel you like. Ideally you'd follow the nixpkgs channel used in the nixvim branch you're using.

That custom nvim package can the be a flake output, and/or included in `home.packages`, `environment.systemPackages`, `users.users.<name>.packages`, etc.

See [Installation (Standalone usage)](https://nix-community.github.io/nixvim/#standalone-usage) and [standalone Usage](https://nix-community.github.io/nixvim/modules/standalone.html) in the docs.

**Reply by gbrls** [reply]
Thanks, this worked like a charm! One issue that I have now is that I'm using this config in multiple computers but I want to add an input the standalone nixvim to optionally disable a set of plugins, but since it's a flake there doesn't seem to be a way to do this.

**Reply by MattSturgeon** [reply]
Essentially, you want to build multiple nvim packages.

You could either

- Use a common config but pass in `extraSpecialArgs` to `makeNivxmiWithModule`, with different values for each variant.
- Build a standard nvim with one config, then _extend_ it with additional config using [`<derivation>.nixvimExtend`](https://nix-community.github.io/nixvim/modules/standalone.html#extending-an-existing-configuration).

When you're done, your flake would output multiple `nvim-*` packages for you to choose from in your system configs.

You could name them something like `nvim-minimal`, `nvim-standard`, `nvim-extra` or whatever you like.

You could also use `nixvimExtend` directly in your system config if the extra config is specific to that system and you won't need that nvim variant elsewhere.
